### ISSUE TITLE
Question about the loss.

### ISSUE THREAD

**Issue by windygoo** [cvlab-stonybrook/SAMPath#4]
There are five foreground classes (other, tumor, stroma, inflammatory and necrosis) in the BCSS dataset, so why the number of classes if set as 6 in the config.BCSS.py file? It seems that the extra class is used to represent the areas to be ignored.

I'm wondering whether the extra class would interfere with the model's learning in the following way:

Line-37 in the losses.py
inputs_softmax = F.softmax(inputs, dim=1)

In my opinion, the number of classes should be set 5 and the ignored mask should serve as a weight mask in loss reduction.

**Comment by jingweizhang-xyz** [comment]
Yes, we did use a extra class. But it did not contribute to the any final loss or metrics as we set "ignore_index = 0" in the main.py Line-36 and ignored_masks in pl_module_sam_seg.py and losses.py Line-30.

We do this to fit the torchmetrics library.

**Comment by windygoo** [comment]
> Yes, we did use a extra class. But it did not contribute to the any final loss or metrics as we set "ignore_index = 0" in the main.py Line-36 and ignored_masks in pl_module_sam_seg.py and losses.py Line-30.
> 
> We do this to fit the torchmetrics library.

Thanks for your detailed reply!
I have aother question on how to calculate the class-wise IoU and Dice scores on the BCSS dataset. I am not familiar with torchmetrics and pytorch-lightning, can I obtain these results by adding a ClassWarpper into the torchmetrics classes (e.g., JaccardIndex, F1Score)?

**Comment by windygoo** [comment]
> If you need the IOU and Dice of a single class, I think you should do that or directly add code in the evaluation part.

Ok, thanks!
